python/cog/server/helpers.py

```diff
@@ -148,3 +159,42 @@ def run(self) -> None:
                     if drain_tokens_seen >= drain_tokens_needed:
                         self.drain_event.set()
                         drain_tokens_seen = 0
+
+X = TypeVar("X")
+
+# functionally this is the exact same thing as aioprocessing but 0.1% the code
+# however it's still worse than just using actual asynchronous io
```

Comment:
What do you mean by "actual asynchronous io"? And if it's better, why don't we do it instead?

Reply:
I mean loop.connect_read_pipe/connect_write_pipe, so that we're actually epolling the pipe instead of blocking

it would require setting up two pipes so that it's duplex, pickling, and a little more clarity than I have about how to correctly pass stuff into the child. previously when I did this I just use asyncio.create_subprocess_exec and passed stuff explicitly; the automatic stuff confuses me